sound/soc/sof/ipc4-control.c

```diff
@@ -173,15 +173,24 @@ static int sof_ipc4_widget_kcontrol_setup(struct snd_sof_dev *sdev, struct snd_s
 	struct snd_sof_control *scontrol;
 	int ret;
 
-	list_for_each_entry(scontrol, &sdev->kcontrol_list, list)
+	list_for_each_entry(scontrol, &sdev->kcontrol_list, list) {
 		if (scontrol->comp_id == swidget->comp_id) {
-			ret = sof_ipc4_set_volume_data(sdev, swidget, scontrol);
-			if (ret < 0) {
-				dev_err(sdev->dev, "%s: kcontrol %d set up failed for widget %s\n",
-					__func__, scontrol->comp_id, swidget->widget->name);
-				return ret;
+			switch (scontrol->info_type) {
+			case SND_SOC_TPLG_CTL_VOLSW:
+			case SND_SOC_TPLG_CTL_VOLSW_SX:
+			case SND_SOC_TPLG_CTL_VOLSW_XR_SX:
+				ret = sof_ipc4_set_volume_data(sdev, swidget, scontrol);
+				if (ret < 0) {
+					dev_err(sdev->dev, "%s: kcontrol %d set up failed for widget %s\n",
+						__func__, scontrol->comp_id, swidget->widget->name);
+					return ret;
+				}
+			default:
+				/* do nothing */
+				continue;
```

Comment:
do you really want to use `continue`? break means break out of switch, but continue mean the next iteration of for loop?

Reply:
There may be more than one kcontrols. So we need use "continue" to go through all the kcontrols.